external/fv3fit/fv3fit/emulation/data/config.py

```diff
@@ -0,0 +1,239 @@
+        return dacite.from_dict(data_class=cls, data=item_info)
+
+    def load_transform_func(self) -> Callable:
+        func = getattr(transforms, self.name)
+        partial_func = func(*self.args, **self.kwargs)
+
+        if not callable(partial_func):
+            raise TypeError(
+                f"Loaded transform for {self.name} from config is no"
+                " longer callable. Check that partial arguments match"
+                " the signature."
+            )
+        return partial_func
+
+
+def _load_transforms(transforms_to_load: Sequence[_TransformConfigItem]):
+
+    loaded_transforms = [
+        transform_info.load_transform_func() for transform_info in transforms_to_load
+    ]
+    logger.debug(
+        f"Loaded transform sequence: {[xfm.name for xfm in transforms_to_load]}"
+    )
+    return compose_left(*loaded_transforms)
+
+
+TransformItemSpec = Union[Mapping, _TransformConfigItem]
+TransformConfigSpec = Dict[str, Sequence[TransformItemSpec]]
+
+
+@dataclasses.dataclass
+class TransformConfig:
+    """
+    Specify a custom transform pipeline for data
+
+    Args
+    ----
+        transforms: Sequence of transform configurations to combine in order
+
+    Example
+    -------
+    Example yaml converting dataset to tensors and grouping::
+
+        transforms:
+          - name: to_tensors
+          - name: group_inputs_outputs
+            args:
+              - ["field1", "field2"]
+              - ["field3"]
+    """
+
+    transforms: Sequence[_TransformConfigItem] = dataclasses.field(default_factory=list)
+
+    @staticmethod
+    def _initialize_custom_transforms(d: TransformConfigSpec):
+        v = d.pop("transforms")
+        initialized = [_TransformConfigItem.from_dict(item_info) for item_info in v]
+        d["transforms"] = initialized
+        return d
+
+    @classmethod
+    def from_dict(cls, d):
+        if "transforms" in d:
+            d = cls._initialize_custom_transforms(d)
+        return dacite.from_dict(data_class=cls, data=d)
+
+    def get_transform_pipeline(self):
+        if self.transforms:
+            return _load_transforms(self.transforms)
+        else:
+            return lambda x: x
+
+
+@dataclasses.dataclass
+class InputTransformConfig(TransformConfig):
+    """
+    Standard input pipeline that goes from xarray dataset to grouped
+    X, y tuples of arrays/tensors per variable
+
+    Args
+    ----
+        input_variables: Variables to include as inputs for training
+        output_variables: Variables to include as targets for training
+        antarctic_only: Limit data to < 60 S.  Requires latitude exists
+            as a field in the dataset
+        use_tensors: Converts data to float32 tensors instead of numpy arrays
+            vertical_subselection: Limit the feature dimension of a variable
+            to a specified range. Loaded in as slices from a 2 or 3 item
+            sequence.
+        transforms: Sequence of extra transform configurations to combine
```

Comment:
This is inherited from the TransformConfig class, but I am open to removing it since there's no immediate use case right now, and the fact that this configuration is not meant for a large amount of flexibility.